src/RemoteSymbolProvider/MicrosoftSymbolServerSymbolProviderIntegrationTest.cpp

```diff
@@ -0,0 +1,102 @@
+#include "OrbitBase/TemporaryFile.h"
+#include "QtUtils/MainThreadExecutorImpl.h"
+#include "RemoteSymbolProvider/MicrosoftSymbolServerSymbolProvider.h"
+#include "SymbolProvider/SymbolLoadingOutcome.h"
+#include "Symbols/MockSymbolCache.h"
+#include "TestUtils/TestUtils.h"
+
+namespace orbit_remote_symbol_provider {
+
+using orbit_grpc_protos::SymbolInfo;
+using orbit_test_utils::HasNoError;
+
+TEST(MicrosoftSymbolServerSymbolProviderIntegrationTest, RetrieveWindowsPdbAndLoadDebugSymbols) {
+  auto temporary_file_or_error = orbit_base::TemporaryFile::Create();
+  ASSERT_THAT(temporary_file_or_error, HasNoError());
+  orbit_base::TemporaryFile temporary_file = std::move(temporary_file_or_error.value());
+  ASSERT_TRUE(temporary_file.file_path().has_parent_path());
+  const std::filesystem::path kSymbolCacheDir = temporary_file.file_path().parent_path();
+
+  orbit_symbols::MockSymbolCache symbol_cache;
+  EXPECT_CALL(symbol_cache, GenerateCachedFilePath)
+      .WillRepeatedly([&kSymbolCacheDir](const std::filesystem::path& module_file_path) {
+        auto file_name = absl::StrReplaceAll(module_file_path.string(), {{"/", "_"}});
+        return kSymbolCacheDir / file_name;
+      });
+
+  orbit_http::HttpDownloadManager download_manager{};
+  MicrosoftSymbolServerSymbolProvider symbol_provider{&symbol_cache, &download_manager};
+
+  std::shared_ptr<orbit_qt_utils::MainThreadExecutorImpl> executor{
+      orbit_qt_utils::MainThreadExecutorImpl::Create()};
+
+  const std::string kValidModuleName{"d3d11.pdb"};
+  const std::string kValidModuleBuildId{"FF5440275BFED43A86CC2B1F287A72151"};
+  orbit_symbol_provider::ModuleIdentifier kValidModuleId{kValidModuleName, kValidModuleBuildId};
+
+  orbit_base::StopSource stop_source{};
+
+  symbol_provider.RetrieveSymbols(kValidModuleId, stop_source.GetStopToken())
+      .Then(executor.get(), [](const orbit_symbol_provider::SymbolLoadingOutcome& result) {
+        ASSERT_TRUE(orbit_symbol_provider::IsSuccessResult(result));
+        orbit_symbol_provider::SymbolLoadingSuccessResult success_result =
+            orbit_symbol_provider::GetSuccessResult(result);
+
+        auto exists_or_error = orbit_base::FileExists(success_result.path);
+        ASSERT_THAT(exists_or_error, HasNoError());
+        EXPECT_TRUE(exists_or_error.value());
+
+        constexpr uint64_t kLoadBias = 0x10000;
+        auto symbols_file_or_error = orbit_object_utils::CreateSymbolsFile(
+            success_result.path, orbit_object_utils::ObjectFileInfo{kLoadBias});
+        ASSERT_THAT(symbols_file_or_error, HasNoError());
+
+        auto symbols_result = symbols_file_or_error.value()->LoadDebugSymbols();
+        ASSERT_THAT(symbols_result, HasNoError());
+
+        absl::flat_hash_map<uint64_t, const SymbolInfo*> symbol_infos_by_address;
+        for (const SymbolInfo& symbol_info : symbols_result.value().symbol_infos()) {
+          symbol_infos_by_address.emplace(symbol_info.address(), &symbol_info);
+        }
+
+        {
```

Comment:
How about also testing the total number of symbols?

Reply:
Done!

But I found `symbo_info`s that have same address http://screenshot/44Zfz7waGkftbPs. Shall I change to use the name rather than the address as the key of the map?

Reply:
Thanks. Yeah. This is fine. There are sometimes multiple names for the same code.